pkg/template/nstemplatetier_test.go

```diff
@@ -0,0 +1,214 @@
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "basic",
+			Namespace: "toolchain-host-operator",
+		},
+	}
+	cl := fake.NewFakeClient(secret, basicTier)
+	service := cluster.KubeFedClusterService{Log: logf.Log, Client: cl}
+	service.AddKubeFedCluster(kubeFedCluster)
+
+	t.Run("success", func(t *testing.T) {
+		// when
+		tmpls, err := template.GetNSTemplates("basic")
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, tmpls, 3)
+	})
+
+	t.Run("failures", func(t *testing.T) {
+
+		t.Run("unknown tier", func(t *testing.T) {
+			// when
+			_, err := template.GetNSTemplates("unknown")
+			// then
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "unable to retrieve the NSTemplateTier 'unknown' from 'Host' cluster")
+		})
+	})
+
+}
+
+func newHostCluster(name string) (*v1beta1.KubeFedCluster, *corev1.Secret) {
+	// make sure that the client gets a valid 200 OK response
+	// when trying to ping the `/api` endpoint on the fake host cluster
+	gock.Observe(gock.DumpRequest)
+
+	gock.New("https://" + name).
+		Get("api").
+		Reply(200).
+		BodyString(`{
+			"kind": "APIVersions",
+			"versions": [
+			  "v1"
+			]
+		  }`)
+	gock.New("https://" + name).
+		Get("api/v1").
+		Reply(200).
+		BodyString(`{
+			"kind": "APIResourceList",
+			"groupVersion": "v1",
+			"resources": [
+			]
+		  }`)
+	gock.New("https://" + name).
+		Get("apis").
+		Reply(200).
+		BodyString(`{
+			"kind": "APIGroupList",
+			"apiVersion": "v1",
+			"groups": [
+			  {
+				"name": "toolchain.dev.openshift.com",
+				"versions": [
+				  {
+					"groupVersion": "toolchain.dev.openshift.com/v1alpha1",
+					"version": "v1alpha1"
+				  }
+				],
+				"preferredVersion": {
+				  "groupVersion": "toolchain.dev.openshift.com/v1alpha1",
+				  "version": "v1alpha1"
+				}
+			  }
+			]
+		  }`)
+	gock.New("https://" + name).
+		Get("apis/toolchain.dev.openshift.com/v1alpha1").
+		Reply(200).
+		BodyString(`{
+			"kind": "APIResourceList",
+			"apiVersion": "v1",
+			"groupVersion": "toolchain.dev.openshift.com/v1alpha1",
+			"resources": [
+				{
+					"name": "nstemplatetiers",
+					"singularName": "nstemplatetier",
+					"namespaced": true,
+					"kind": "NSTemplateTier",
+					"verbs": [
+						"delete",
+						"deletecollection",
+						"get",
+						"list",
+						"patch",
+						"create",
+						"update",
+						"watch"
+					]
+				},
+				{
+					"name": "nstemplatetiers/status",
+					"singularName": "",
+					"namespaced": true,
+					"kind": "NSTemplateTier",
+					"verbs": [
+						"get",
+						"patch",
+						"update"
+					]
+				}
+			]
+		}`)
+	gock.New("https://" + name).
+		Get("/apis/toolchain.dev.openshift.com/v1alpha1/namespaces/toolchain-host-operator/nstemplatetiers/basic").
+		Reply(200).
+		BodyString(`{
+			"apiVersion": "toolchain.dev.openshift.com/v1alpha1",
+			"kind": "NSTemplateTier",
+			"metadata": {
+				"name": "basic",
+				"namespace": "host-operator-1568102314"
+			},
+			"spec": {
+				"namespaces": [
```

Comment:
The biggest drawback of this approach that I see is that it will be hard to maintain in case we change anything in the structure of the `NSTemplateTier` type

Reply:
outdated since it was removed in https://github.com/codeready-toolchain/member-operator/commit/21bb803d4a1d43875aabaf0f26d516f94a817ad2